dspace-api/src/test/java/org/dspace/app/util/SubmissionConfigTest.java

```diff
@@ -87,4 +109,35 @@ public void testReadAndProcessTypeBindSubmissionConfig()
         assertEquals(unboundFields, allowedFieldsForArticle);
         assertEquals(unboundFields, allowedFieldsForNoType);
     }
+
+    @Test
+    public void testSubmissionMapByCommunityHandleSubmissionConfig()
+        throws SubmissionConfigReaderException, DCInputsReaderException, SQLException {
+
+        // Sep up a structure with one top community and two subcommunities
+        // with one collection
+        when(col1.getHandle()).thenReturn("123456789/not-mapped1");
+        when(col1.getCommunities()).thenReturn(List.of(subcom1));
+
+        when(col2.getHandle()).thenReturn("123456789/not-mapped2");
+        when(col2.getCommunities()).thenReturn(List.of(subcom2));
+
+        when(subcom1.getHandle()).thenReturn("123456789/subcommunity-test");
+
+        when(subcom2.getParentCommunities()).thenReturn(List.of(topcom));
+        when(subcom2.getHandle()).thenReturn("123456789/not-mapped3");
+
+        when(topcom.getHandle()).thenReturn("123456789/topcommunity-test");
+
+        // for col1, it should return the item submission form defined for their parent subcom1
+        SubmissionConfig submissionConfig1 =
+            new SubmissionConfigReader().getSubmissionConfigByCollection(col1);
+        assertEquals("subcommunitytest", submissionConfig1.getSubmissionName());
+
+        // for col2, it should return the item submission form defined for topcom
+        SubmissionConfig submissionConfig2 =
+            new SubmissionConfigReader().getSubmissionConfigByCollection(col2);
+        assertEquals("topcommunitytest", submissionConfig2.getSubmissionName());
+
```

Comment:
should we check that a collection with a dedicated submission form within the topcomm get the collection specific form instead than the topcomm one?

Reply:
we also need to test the respect of the precedence of the entity mapping over the community mapping